reference/FungibleToken.scilla

```diff
@@ -0,0 +1,394 @@
+  | Some bal =>
+    get_spender_allowed <- allowances_map[from][_sender];
+    match get_spender_allowed with
+    | None =>
+      err = CodeNotAuthorized;
+      MakeErrorEvent err
+    | Some allowed =>
+      min = min_int bal allowed;
+      can_do = uint128_le amount min;
+      match can_do with
+      | True =>
+        ProcedureMove from to amount;
+        new_allowed = builtin sub allowed amount;
+        allowances_map[from][_sender] := new_allowed
+      | False =>
+        err = CodeNotAllowed;
+        MakeErrorEvent err
+      end
+    end
+  end
+end
+
+(* @dev: Optional transition. Mint new tokens. Only contract_owner can mint. *)
+(* @param recipient: Address of the recipient whose balance is to increase.  *)
+(* @param amount:    Number of tokens to be minted.                          *)
+transition Mint(recipient: ByStr20, amount: Uint128)
+  is_owner = builtin eq _sender contract_owner;
+   match is_owner with
+    | False =>
+      err = CodeNotAuthorized;
+      MakeErrorEvent err
+    | True =>
+      ProcedureMint recipient amount
+    end
+end
+
+(* @dev: Optional transition. Burn existing tokens. Only contract_owner can burn. *)
+(* @param burn_account: Address of the token_owner whose balance is to decrease.  *)
+(* @param amount:       Number of tokens to be burned.                            *)
+transition Burn(burn_account: ByStr20, amount: Uint128)
+  is_owner = builtin eq _sender contract_owner;
+    match is_owner with
+    | False =>
+      err = CodeNotAuthorized;
+      MakeErrorEvent err
+    | True =>
+      ProcedureBurn burn_account amount
+    end
+end
+
+(* @dev: Make an address an operator of the caller.             *)
+(* @param operator: Address to be authorize as operator or      *)
+(* re-authorize as default_operator. Cannot be calling address. *)
+transition AuthorizeOperator(operator: ByStr20)
+  is_sender = builtin eq operator _sender;
+  match is_sender with
+  | True => 
+    (* _sender is authorizing self as operator, return error code *)
+    err = CodeNotAuthorized;
+    MakeErrorEvent err
+  | False =>
+    is_default_operator =  is_default_operator f_eq operator default_operators;
+    match is_default_operator with
+    | True =>
+      (* Re-authorize default_operator *)
+      delete revoked_default_operators[_sender][operator];
+      e = { _eventname : "ReAuthorizedDefaultOperatorSuccess"; reauthorized_default_operator : operator; authorizer : _sender};
+      event e
+    | False =>
+      (* Authorize new operator *)
+      authorize = True;
+      operators_map[_sender][operator] := authorize;
+      e = {_eventname : "AuthorizeOperatorSuccess"; authorized_operator : operator; authorizer : _sender};
+      event e
+    end
+  end
+end
+
+(* @dev: Revoke an address from being an operator or default_operator of the caller. *)
+(* @param operator: Address to be removed as operator or default_operator.           *)
+transition RevokeOperator(operator: ByStr20)
```

Comment:
For `AuthorizeOperator` there is a check to see if the sender is trying to add themselves into the authorized list. However for the `RevokeOperator`, there is no such check if the sender is trying to add themselves into the revoked list?

Reply:
Good suggestion, adding the check for self.